### PR TITLE
Allow ignoring warnings from configuration.

### PR DESCRIPTION
**Pull request by blimmer**
Since `--quiet` is [not a supported configuration](https://github.com/eslint/eslint/issues/1768) for the CLIEngine, I think this is how we'll have to address #142 .

Outstanding Questions:
* [x] manual testing - I could find examples of how to test this locally against a project. are there docs for this or can someone point me in the right direction?

~~* [ ] automated testing - is there any that can be added for this?~~
Integration testing seems out of scope for this change
~~* [ ] docs - where should this config option be documented?~~
docs website needs to be updated

### CODE REVIEW

File: bin/eslint.js [hunk 1]
```diff
@@ -209,6 +214,8 @@ function analyzeFiles() {
         var path = result.filePath.replace(/^\/code\//, "");
 
         result.messages.forEach(function(message) {
+          if (ignoreWarnings && message.severity === 1) { return; }
```
Comment (dblandin) [inline bin/eslint.js]: I'm assuming that `1` maps to the warning level but can we explicitly map this value to a named variable? Maybe up top as `WARNING_LEVEL`?

**Comment by dblandin** [comment]
> manual testing - I could find examples of how to test this locally against a project. are there docs for this or can someone point me in the right direction?

You can test your engine update locally with [the CLI](https://github.com/codeclimate/codeclimate). You can either override the currently downloaded image for the `eslint-3` channel or specify your own local engine name:

```bash
$ IMAGE_NAME=codeclimate/codeclimate-eslint:eslint-3 make image
```

```yaml
eslint:
  enabled: true
  channel: eslint-3
```

```bash
$ codeclimate analyze
```

```
$ IMAGE_NAME=codeclimate/codeclimate-eslint-quiet make image
```

```yaml
eslint-quiet:
  enabled: true
```

```bash
$ codeclimate analyze --dev
```

> automated testing - is there any that can be added for this?

There are a few unit level specs in this repo but nothing that exercises the whole engine. I'd be happy with manual testing, but if you're willing to implement an integration spec, either in this PR or a following PR, I'd gladly review it :)

> docs - where should this config option be documented?

This engine should really have a "Usage" section in the README which includes all of the user config options. At the moment, we have more thorough documentation on our docs site: https://docs.codeclimate.com/docs/eslint

**Comment by blimmer** [comment]
@dblandin this is ready for review again. RE the previous feedback, I've constant-ized the warning level as request and run some manual testing proving that this is working correctly.

With quiet turned off on one of my projects:
```
Analysis complete! Found 523 issues.
```

And with this config:
```yml
  eslint-quiet:
    enabled: true
    config:
     ignore_warnings: true
```

I get this (which is only reporting error levels eslint findings):
```
Analysis complete! Found 8 issues.
```

**Comment by blimmer** [comment]
Also, I added a [CONTRIBUTING.md](https://github.com/codeclimate/codeclimate-eslint/pull/144/commits/05e63bb1b0a656dccf98fe9ca5f7a80f76f88e37) file with the instructions you gave me to test locally against the CLI.

**Comment by blimmer** [comment]
bump @dblandin - anything blocking this PR?

**Comment by dblandin** [comment]
This looks great to me! We can address the README usage instructions in a separate PR.

Thanks!

**Comment by blimmer** [comment]
Awesome, thanks. Very excited to see this feature go out. Is the README change something I can help with? Seems like it needs to be documented in a private repo for the docs site?

**Comment by dblandin** [comment]
To update the usage instructions on our docs site, you can use the "Suggest Edits" feature at https://docs.codeclimate.com/docs/eslint. I'd also welcome a PR to this repository's README documenting the configuration option. The docs site usage instructions tend to be a bit broader than what we cover in engine READMEs.

**Pull request merged**